7/ The table below provides a training data set containing six observations, three predictors, and one qualitative response variable.

In [5]:
import numpy as np

obs1 = np.array([0,3,0])
obs2 = np.array([2,0,0])
obs3 = np.array([0,1,3])
obs4 = np.array([0,1,2])
obs5 = np.array([-1,0,1])
obs6 = np.array([1,1,1])

test = np.array([0,0,0])

a/ Compute the Euclidean distance between each observation and the test point, X1 = X2 = X3 = 0.

In [7]:
eucl1 = np.linalg.norm(test - obs1)
eucl1

3.0

In [8]:
eucl2 = np.linalg.norm(test - obs2)
eucl2

2.0

In [9]:
eucl3 = np.linalg.norm(test - obs3)
eucl3

3.1622776601683795

In [10]:
eucl4 = np.linalg.norm(test - obs4)
eucl4

2.23606797749979

In [11]:
eucl5 = np.linalg.norm(test - obs5)
eucl5

1.4142135623730951

In [12]:
eucl6 = np.linalg.norm(test - obs6)
eucl6

1.7320508075688772

b/ What is our prediction with K = 1? Why?

For K=1, the test point is Green. Since Observation 5 with the value Y of "Green" has the Euclidean distance of 1.414, which is the closest to K value of 1.

c/ What is our prediction with K = 3? Why?

For K=3, the test point could be Red. Since the closest value to K=3 are Observation 2, 5, and 6, where 2 has a Y value of "Green", and "Red" for both 5 and 6.

d/ If the Bayes decision boundary in this problem is highly nonlinear, then would we expect the best value for K to be large or small? Why?

If the Bayes decision boundary in this problem is highly nonlinear, the Bayes classifier has a low bias by very high variance. This implies that the we can expect the best value for K to be smaller.